core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -253,11 +260,17 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     timeout.max(member.rebalanceTimeoutMs)
   }
 
-  // TODO: decide if ids should be predictable or random
   def generateMemberIdSuffix = UUID.randomUUID().toString
 
   def canRebalance = GroupMetadata.validPreviousStates(PreparingRebalance).contains(state)
 
+  def protocolNotMatch(memberProtocolType: String,
```

Comment:
Would it be possible to consolidate this logic into `supportsProtocols`? Also, I think it might be simpler to do a simple `if (is(Empty))` check and split the two cases.

Reply:
Simplified the logic by using isEmpty to check the two cases. And converged two checks.